system-tests/test/utils/context.ts

```diff
@@ -0,0 +1,143 @@
+import fs from "fs"
+import { helpers, network } from "hardhat"
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers"
+import { KeyPair as BitcoinKeyPair, keyPairFromPrivateWif } from "./bitcoin"
+
+// TODO: For now, the context and its setup is global and identical for each
+//       scenario. Once more scenarios is added, this should be probably
+//       split into the global common context and specialized per-scenario addons.
+
+/**
+ * Represents a context of the given system tests scenario.
+ */
+export interface SystemTestsContext {
+  /**
+   * Electrum instance connection URL.
+   */
+  electrumUrl: string
+  /**
+   * Handle to the contracts' deployment info.
+   */
+  contractsDeploymentInfo: ContractsDeploymentInfo
+  /**
+   * Ethereum signer representing the contract governance.
+   */
+  governance: SignerWithAddress
+  /**
+   * Ethereum signer representing the system maintainer.
+   */
+  maintainer: SignerWithAddress
+  /**
+   * Ethereum signer representing the depositor.
+   */
+  depositor: SignerWithAddress
+  /**
+   * Bitcoin key pair of the depositor.
+   */
+  depositorBitcoinKeyPair: BitcoinKeyPair
+  /**
+   * Bitcoin key pair of the wallet.
+   */
+  walletBitcoinKeyPair: BitcoinKeyPair
+}
+
+/**
+ * Contracts deployment info that contains deployed contracts' addresses and ABIs.
+ */
+interface ContractsDeploymentInfo {
+  contracts: {
+    [key: string]: {
+      address: string
+      abi: any
+    }
+  }
+}
+
+/**
+ * Sets up the system tests context.
+ * @returns System tests context.
+ */
+export async function setupSystemTestsContext(): Promise<SystemTestsContext> {
+  const electrumUrl = process.env.ELECTRUM_URL
+  if (!electrumUrl) {
+    throw new Error(`ELECTRUM_URL is not set`)
+  }
+
+  if (network.name === "hardhat") {
+    throw new Error("Built-in Hardhat network is not supported")
+  }
+
+  const contractsDeploymentInfo = readContractsDeploymentExportFile()
```

Comment:
See https://github.com/keep-network/tbtc-v2/pull/339/commits/fe63e613f8f649fd2055c3b17b94e5c5b780587f. However, I used just the `name` validation since `chainId` is not always set. I think it should be enough to exclude mistakes related to the wrong deployment export file.